### DISCUSSION TITLE
duplicate names

### DISCUSSION THREAD

**Discussion by asliyurt** [faker-js/faker#1734] [Q&A]
hey, i am using faker to generate unique names in tests. It works fine when tests run sequentially, however when i run the things concurrently, it causes an issue as it gets the exact same name for each test running at the same time, where the names should be unique. is there a way to avoid this for parallel tests? Thanks

**Comment by ST-DDT** [comment]
Have you tried using our unique method?
https://fakerjs.dev/api/helpers.html#unique

Also which version of faker are you using?

**Reply by Shinigami92** [reply]
I don't think this will solve it

I assume @asliyurt need to use `faker.seed(...)` in combination with `beforeAll`
Some test-runners run tests in isolated workers and so the `unique` function would not share its store
In vitest you could also maybe change the random sequence.seed https://vitest.dev/config/#sequence-seed

**Reply by ST-DDT** [reply]
> I assume @asliyurt need to use `faker.seed(...)` in combination with `beforeAll`.

This only has only a chance to fix the issue if they are using an old version of faker (pre [v7.6.0](https://github.com/faker-js/faker/commit/925db3a0194fba6d5984203a69747265848227ef)), that initialized the seed from the current time.
Calling seed with any value, does not prevent/reduce collisions otherwise.

---

We are unable to prevent collisions from isolated workers/processes. If you find an easy solution, please let us know.
(IMO isolated workers/processes should either not be able to cause collisions for tests (due to the test setup) or should be able to detect and ignore these errors e.g. via special error code)
How would you handle the duplicate data in the real world? E.g. Users either forget that they already have an account or common names (such as Mr. Smith) are very likely to cause collisions as well.
